api/src/index.ts

```diff
@@ -14,75 +14,34 @@
  * limitations under the License.
  */
 
-export * from './baggage/types';
-export { baggageEntryMetadataFromString } from './baggage/utils';
-export * from './common/Exception';
-export * from './common/Time';
-export * from './common/Attributes';
-export * from './diag';
-export * from './propagation/TextMapPropagator';
-export * from './trace/attributes';
-export * from './trace/link';
-export * from './trace/ProxyTracer';
-export * from './trace/ProxyTracerProvider';
-export * from './trace/Sampler';
-export * from './trace/SamplingResult';
-export * from './trace/span_context';
-export * from './trace/span_kind';
-export * from './trace/span';
-export * from './trace/SpanOptions';
-export * from './trace/status';
-export * from './trace/trace_flags';
-export * from './trace/trace_state';
-export { createTraceState } from './trace/internal/utils';
-export * from './trace/tracer_provider';
-export * from './trace/tracer';
-export * from './trace/tracer_options';
-
-export {
-  isSpanContextValid,
-  isValidTraceId,
-  isValidSpanId,
-} from './trace/spancontext-utils';
-
 export {
-  INVALID_SPANID,
-  INVALID_TRACEID,
-  INVALID_SPAN_CONTEXT,
-} from './trace/invalid-span-constants';
-
-export * from './context/context';
-export * from './context/types';
-
-import { ContextAPI } from './api/context';
-export type { ContextAPI } from './api/context';
-/** Entrypoint for context API */
-export const context = ContextAPI.getInstance();
-
-import { TraceAPI } from './api/trace';
-export type { TraceAPI } from './api/trace';
-/** Entrypoint for trace API */
-export const trace = TraceAPI.getInstance();
-
-import { PropagationAPI } from './api/propagation';
-export type { PropagationAPI } from './api/propagation';
-/** Entrypoint for propagation API */
-export const propagation = PropagationAPI.getInstance();
-
-import { DiagAPI } from './api/diag';
-export type { DiagAPI } from './api/diag';
-
-/**
- * Entrypoint for Diag API.
- * Defines Diagnostic handler used for internal diagnostic logging operations.
- * The default provides a Noop DiagLogger implementation which may be changed via the
- * diag.setLogger(logger: DiagLogger) function.
- */
-export const diag = DiagAPI.instance();
+  BaggageEntry,
+  BaggageEntryMetadata,
+  Baggage,
+} from './baggage/types';
+export { baggageEntryMetadataFromString } from './baggage/utils';
+export { Exception } from './common/Exception';
+export { HrTime, TimeInput } from './common/Time';
+export { Attributes, AttributeValue } from './common/Attributes';
+
+export * from './context-api';
```

Comment:
It would be nice to have this as explicit (named) exports rather than implicit exports (*) to remove the unexpected inclusion of exported values from the listed files.

Reply:
I have updated the exports in *-api.ts file to be explicit so I just export everything from *-api.ts from index.ts. Like trace-api.ts, which used to be `export *` from each internal module.

Reply:
I agree with @MSNev here. The index is the one source of truth to see what is exported from the API package. It is not convenient to have to go to another file to see what is exported. Also, someone may add an export to that file not realizing that index.ts has the `*` export.